### DISCUSSION TITLE
Roboflow support

### DISCUSSION THREAD

**Discussion by SkalskiP** [fcakyon/yolov5-pip#209] [Ideas]
Hi, @fcakyon 👋🏻!

Would you mind if I prepared a PR adding support for Roboflow datasets to your pip package?

**Comment by fcakyon** [comment]
Hello @SkalskiP it would be great! Feel free to open such PR 👍🏻

**Comment by SkalskiP** [comment]
Hi, @fcakyon 👋🏻! It took me a bit longer than expected, but here is the first PR that allows direct use of datasets from Roboflow Universe to train models. Let me know what you think. 

https://github.com/fcakyon/yolov5-pip/pull/210

**Reply by fcakyon** [reply]
@SkalskiP, thanks for the PR! I have merged your PR to run roboflow cli tests, but there are several issues.

1. when the target dataset type is selected as `yolov8`, roboflow package requires ultralytics package to be installed in the env (which is not a yolov5-pip dependency): https://github.com/fcakyon/yolov5-pip/actions/runs/4022619564/jobs/6912550070
2. when the target dataset type is selected as `yolov5`, roboflow API returns a 404 error: https://github.com/fcakyon/yolov5-pip/actions/runs/4022749506/jobs/6912813939

Would you happen to have any advice to fix these issues?

PS: I am using this dataset in roboflow tests since its small: https://universe.roboflow.com/jacob-solawetz/aerial-maritime/dataset/10

**Reply by SkalskiP** [reply]
Hi, @fcakyon! I didn't see those issues yesterday. But of course, that needs to work flawlessly so you can count on my help. I'll start to work on a fix right away.

**Reply by SkalskiP** [reply]
Thank you very much for your help with CI. That is awesome that we can test automatically. 😎

**Reply by SkalskiP** [reply]
Hi, @fcakyon 👋🏻! 

1. There is a minor bug in my PR that I'm going to fix. I always meant to download the dataset in `yolov5` format for `detection`, not in `yolov8`.
2. It Looks like we have two bugs that you accidentally uncovered. I already created issues on our side:
- https://github.com/roboflow/roboflow-python/issues/103
- https://github.com/roboflow/roboflow-python/issues/104

Interestingly that 404 error is thrown for https://universe.roboflow.com/jacob-solawetz/aerial-maritime/dataset/10 but not for other datasets I tested. That's why I propose using a different dataset in the CI/CD pipeline and let us debug that in parallel.

**Comment by SkalskiP** [comment]
Hi, @fcakyon 👋🏻! 

I created the fix: https://github.com/fcakyon/yolov5-pip/pull/213. I tested all your CI/CD cases locally, and they all work. Your test suit fails on my PR now because I don't have access to your  `env.ROBOFLOW_TOKEN`, and it looks like it's not being passed. :/ But it should work for you.

**Reply by fcakyon** [reply]
Ok lets merge and see how tests perform. Thanks for your efforts 👍

**Reply by SkalskiP** [reply]
Thanks for the opportunity and responsiveness.

**Reply by SkalskiP** [reply]
And yes, lets marge run CI. I'm happy to assist until it will work obviously :)

**Comment by fcakyon** [comment]
@SkalskiP it gives such error: https://github.com/fcakyon/yolov5-pip/actions/runs/4025808461/jobs/6919504263

**Reply by SkalskiP** [reply]
Do you think it's related to my changes?

**Reply by fcakyon** [reply]
After checking the workflow logs, roboflow detection training finishes successfully. Then it fails during classification model training. 

It may be related to dataset or ultralytics repo implementation. I will check it once I am back at home 👍🏻

**Reply by SkalskiP** [reply]
Yup looks like detection works. But classification fails, and it fails after training is completed

**Reply by SkalskiP** [reply]
I did some digging. Maybe it'll help you to debug it. So it looks like the problem originates in `imshow_cls` function. Values in `pred` tensor should come from `0` to `N-1` where `N` is the number of classes in the dataset.

**Reply by fcakyon** [reply]
I faced a similar issue with one of the roboflow datasets when there were duğlicate classes (same class name was present 2 times in the nc field of the data.yaml file). Maybe a similar issue is present here.

**Reply by SkalskiP** [reply]
Hi, @fcakyon 👋🏻 Did you find time to look at the classification issues?

**Reply by fcakyon** [reply]
I have started looking but not sure on how to fix it.

**Reply by SkalskiP** [reply]
I was thinking: maybe we could focus on detection and segmentation for now? Make sure that those two work, and then add classification on top of that.

**Reply by fcakyon** [reply]
Yes that would be better 👍  https://github.com/fcakyon/yolov5-pip/pull/215/commits/cacab0cc3943bfe6c085946a3aea243a59a757a4

**Reply by fcakyon** [reply]
I will merge and publish a release once tests pass: https://github.com/fcakyon/yolov5-pip/actions/runs/4058091434

**Reply by SkalskiP** [reply]
great! <3 Than I was thinking about adding deploy to roboflow

**Reply by fcakyon** [reply]
Nice! New version is live: https://github.com/fcakyon/yolov5-pip/releases/tag/7.0.8

**Reply by SkalskiP** [reply]
🚀 I'll share the news on LinkedIn tomorrow!

**Reply by SkalskiP** [reply]
@fcakyon btw, are you still in the US?

**Reply by fcakyon** [reply]
I am currently in Turkey.
